### ISSUE TITLE
Combine marking bar and scroll bar

### ISSUE THREAD

**Issue by sphh** [spyder-ide/spyder#16193]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

Marking for occurrences of the search string, linting warnings, TODO/BUG/FIXME items are displayed in right border of the editor pane (between the text field and the scroll bar). This is very helpful and I use them quite a lot!

I also noticed that you can click in this area and it scrolls to the relevant part in the editor. This doubles functionality of the scroll bar (at least in Linux, where I can click with the right mouse button in the scroll bar and the editor takes me to the absolute position – exactly as left clicking in the ‘marking strip’ does).

This left me wondering, if it is possible to combine those two scroll bars? For example Firefox overlays the occurrences of the search string onto the scroll bar … This would be highly welcome on small screens, where ‘every pixel helps’.

## Versions

* Spyder version: 5.1.1
* Python version: 3.8.10 64-bit
* Qt version: 5.12.9
* PyQt version: 5.12.3
* Operating System name/version: Linux Mint 20.2 Cinnamon

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.26.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.0.1 (OK)
nbconvert >=4.0               :  6.1.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
parso >=0.7.0;<0.9.0          :  0.7.0 (OK)
pexpect >=4.4.0               :  4.6.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.5.1 (OK)
pygments >=2.0                :  2.9.0 (OK)
pylint >=2.5.0                :  2.9.6 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.1;<1.3.0          :  1.2.1 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.1.0             :  5.1.1 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  57.4.0 (OK)
sphinx >=0.6.6                :  4.1.2 (OK)
spyder_kernels >=2.1.0;<2.2.0 :  2.1.0 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  1.0.2 (OK)
xdg >=0.26                    :  0.26 (OK)
zmq >=17                      :  22.2.0 (OK)

# Optional:
cython >=0.21                 :  None (OK)
matplotlib >=2.0.0            :  3.4.2 (OK)
numpy >=1.7                   :  1.21.1 (OK)
pandas >=1.1.1                :  1.3.1 (OK)
scipy >=0.17.0                :  1.7.1 (OK)
sympy >=0.7.3                 :  1.8 (OK)
```

**Comment by ccordoba12** [comment]
Hey @sphh, thanks for reporting. You said:

> This left me wondering, if it is possible to combine those two scroll bars?

There are two things about this:

* Technically it's not simple to merge those two things into one.
* We discussed that a while ago (see #5085 and #5215) and decided to leave the two of them. You can read the rationale for our decision there.

Let us know if there's something else you'd like to discuss about this topic.

**Comment by sphh** [comment]
Oops, sorry I missed that issue and PR! But at least I now know the right nomenclature.

Reading through the issue, I see that the current implementation is a big improvement! Thanks.

I don't know much about theming and its actual implementation etc., but I would like the following behaviour (as I said, I don't know, if that is possible):

- The flags are displayed inside the (native) scrollbar, stretching from the left to the right of the actual scrollarea.
- The flags become lighter in colour, if the scroll handle is placed over the flags: 
![Untitled](https://user-images.githubusercontent.com/4856462/129012279-aefd9b6d-8d95-4f9f-a74f-7a19a460cbc7.png)

That way it should work with all scroll bars and looks quite neat (because you are perfectly right, that just drawing it over the whole width of the toolbar looks awful).

**Comment by ccordoba12** [comment]
I agree this would be nice to have but for now we're really busy with other things.

**Comment by sphh** [comment]
@ccordoba12: Glad, that you like the idea and putting it on the wishlist. I can perfectly understand, that it is certainly not of the highest priority.
